### PR TITLE
refactor: consolidate BigQuery client creation and set user-agent

### LINKED ISSUE #98: Include sqlalchemy/VERSION in the user-agent string when constructing BQ client(s)
**Is your feature request related to a problem? Please describe.**

I'd like to be able to attribute requests from this library. This is useful for measuring the impact of this project and helping me prioritize engineering resources across our open source connectors.

 **Describe the solution you'd like**

Construct a [ClientInfo](https://googleapis.dev/python/google-api-core/latest/client_info.html) object and pass to the `bigquery.Client` constructor whereever we use it.

 **Describe alternatives you've considered**

N/A

 **Additional context**

* Ibis implementation: https://github.com/ibis-project/ibis/blob/165f78de8f4f0121ba2c601b5c9f89bc0f65a593/ibis/backends/bigquery/client.py#L55-L62 and https://github.com/ibis-project/ibis/blob/165f78de8f4f0121ba2c601b5c9f89bc0f65a593/ibis/backends/bigquery/client.py#L407-L411
* Pandas implementation: https://github.com/pydata/pandas-gbq/blob/22a6064ee616fbdd14ce2c8bf8bfe1ed7d3b6291/pandas_gbq/gbq.py#L401-L428

### PR DESCRIPTION
**Pull request by tswast**
Creating as draft, as I want to add unit tests first.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-sqlalchemy/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #98 🦕

### CODE REVIEW

File: pybigquery/_helpers.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        credentials = credentials.with_scopes(SCOPES)
+        default_project = credentials.project
+    else:
+        credentials, default_project = google.auth.default(scopes=SCOPES)
+
+    if project_id is None:
+        project_id = default_project
+
+    return bigquery.Client(
+        client_info=google_client_info(),
```
Comment (tswast) [inline pybigquery/_helpers.py:55]: Need to add hook to check if `client_info` is available before trying to add it.

https://github.com/pydata/pandas-gbq/blob/853f7922a0e0c853ed4d295ff14c41bfedf7e8d1/pandas_gbq/gbq.py#L364-L369

Comment (tswast) [reply pybigquery/_helpers.py:55]: Actually, 1.12.0 was released 2019-05-16. That's close enough to the NEP 29 recommended 2 years, that I think it's worth just bumping the minimum version of `google-cloud-bigquery` in this case.

**Review by shollyman — APPROVED** [review]
minor comment: I feel like I've seen/heard multiple conversations about resolving default project recently.  I'd love it if this is going to go the route of ADC that we're consistent about the lookup order across languages and clients.

**Comment by tswast** [comment]
@shollyman re: default project lookup, most of the hard work is done by `google.auth.default`. That should stay consistent with the ADC spec.

https://github.com/googleapis/python-bigquery-sqlalchemy/pull/100/files#diff-d0e22c51d2de44de91fbaefb7a9daa158cd6ae091e840a51984a8a3c05d8bb58R49

We don't hit that line if someone specifies a service account key, in which case we use the project ID from the JSON key file. That's consistent with what the `google-cloud-core` library does

https://github.com/googleapis/python-cloud-core/blob/1c9e969f8b05098e1b91ea71d62ca141e2304a6c/google/cloud/client.py#L85

**Pull request merged**